project/Dependencies.scala

```diff
@@ -27,4 +27,41 @@ object Dependencies {
     }
   lazy val scalaXml = scala211Module("scala-xml", "1.0.1")
   lazy val scalaParsers = scala211Module("scala-parser-combinators", "1.0.1")
+
+  // Maven related dependnecy craziness
+  //val mvnEmbedder = "org.apache.maven" % "maven-embedder" % mvnVersion
+  val mvnWagonVersion = "2.4"
+  val mvnVersion = "3.2.3"
+  val aetherVersion = "1.0.1.v20141111"
+
+  val mvnAether = "org.apache.maven" % "maven-aether-provider" % mvnVersion
+  val aether = "org.eclipse.aether" % "aether" % aetherVersion
+  val aetherImpl = "org.eclipse.aether" % "aether-impl" % aetherVersion
+  val aetherUtil = "org.eclipse.aether" % "aether-util" % aetherVersion
+  val aetherTransportFile = "org.eclipse.aether" % "aether-transport-file" % aetherVersion
+  val aetherTransportWagon = "org.eclipse.aether" % "aether-transport-wagon" % aetherVersion
+  val aetherTransportHttp = "org.eclipse.aether" % "aether-transport-http" % aetherVersion
+  val aetherConnectorBasic = "org.eclipse.aether" % "aether-connector-basic" % aetherVersion
+  val sisuPlexus = ("org.eclipse.sisu" % "org.eclipse.sisu.plexus" % "0.3.0.M1").exclude("javax.enterprise", "cdi-api").exclude("com.google.code.findbugs", "jsr305")
+  val guice = "com.google.inject" % "guice" % "3.0"
+  val guava = "com.google.guava" % "guava" % "18.0"
+  val javaxInject = "javax.inject" % "javax.inject" % "1"
```

Comment:
What about coding some scaffolding in chain resolver, but largely moving `MavenRemoteRepositoryResolver` code into a plugin? I am not sure if it's necessarily a good idea, but is that a viable approach?